cabal-install/TESTING.md

```diff
@@ -0,0 +1,24 @@
+# Testing `cabal-install`
```

Comment:
Should this be merged with CONTRIBUTING.md#running-tests?

Reply:
Your folks' preference. I'd think to look in the project first for project-level testing details, but that's just me.

Reply:
I agree it makes sense to have detailed info here. I'd put a link to this file in the testing section of CONTRIBUTING though, and mention cabal-testsuite here too

Reply:
Done